File: JavaScript/7js 3/sumAverage.js
/*
|--------------------------------------------------------------------------
| Sum of Array Averages
|--------------------------------------------------------------------------
|
*/

function sumAverage(arr) {
   res = 0
   arr.forEach(i => {
      res += i.reduce((a, b) => a + b) / i.length;
   });
   return Math.floor(res)
}


function sumAverage(arr) {
   return Math.floor(arr.reduce((pre, val) => pre + val.reduce((p, v) => p + v) / val.length, 0));
}


function sumAverage(arr) {
   return Math.floor(arr.map((a => a.reduce((c, d) => c + d) / a.length)).reduce((j, k) => j + k))
}


console.log(sumAverage([
   [3, 4, 1, 3, 5, 1, 4],
   [21, 54, 33, 21, 77]
]), 44);
console.log(sumAverage([
   [-4, 3, -8, -2],
   [2, 9, 1, -5],
   [-7, -2, -6, -4]
]), -6);

File: Python/7py 3/sum_average.py
"""
|--------------------------------------------------------------------------
| Sum of Array Averages
|--------------------------------------------------------------------------
|
"""

from math import floor
def sum_average(arr):
   return floor(sum([sum(i)/len(i) for i in arr]))


def sum_average(x):
   return sum(sum(i)/len(i) for i in x) // 1


def sum_average(arr):
   avg = 0
   for lst in arr:
      avg += sum(lst) / len(lst)
   return floor(avg)


print(sum_average([[1, 2, 2, 1], [2, 2, 2, 1]]), 3)